include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -244,6 +248,111 @@ make sure that environment variable OPENPMD_BP_BACKEND is not set to ADIOS1.
                    "respectively.",
                    "doubleBuffer"};
 
+            /*
+             * The openPMD plugin is used as a normal I/O plugin as well as for
+             * the creation of checkpoints.
+             * Some parameters should only be available in either of both modes.
+             */
+            enum class ApplyParameter
+            {
+                Always,
+                NotInCheckpoint,
+                OnlyInCheckpoint
+            };
+
+            /*
+             * Information used for configuring a parameter of the openPMD
+             * plugin. Used in the `parameters` member.
+             * Emplacing such a parameter in that list will activate
+             * the parameter in the PIConGPU command line and in the TOML
+             * configuration of the openPMD plugin.
+             */
+            struct Parameter
+            {
+                // Pointer to a command line option object that should be registered
+                plugins::multi::Option<std::string>* commandLineParameter;
+                // Optionally, the name for the parameter in the TOML configuration
+                std::optional<std::string> tomlParameter;
+                // Optionally, a member pointer inside the PluginParameters struct
+                // where the value of this parameter should be read to
+                // If not specifying this, the value must be manually read
+                // (used for the source parameter since the simple logic
+                //  is not sufficient for it)
+                std::optional<std::string PluginParameters::*> targetInConfigObject;
+                // See description of enum class ApplyParameter
+                ApplyParameter applyParameter;
+                // Optionally the additionalDescription parameter of
+                // commandLineParameter->registerHelp()
+                std::optional<std::function<std::string()>> additionalDescription;
+
+                Parameter(
+                    plugins::multi::Option<std::string>* commandLineParameter_in,
+                    std::optional<std::string> tomlParameter_in,
```

Comment:
IMO there is no need for this parameter because the value can be derived with `commandLineParameter_in->getName()` in line 295 and would avoid that we use different names in toml and on the command line.

Reply:
The parameter names are not exactly the same. I used `backend_config` instead of `.json` in the TOML configuration.
`--openPMD.json` is still somewhat descriptive, but `json = "..."` in a TOML file was a bit weird. 
Also for some reason TOML uses `data_preparation_strategy`, but the command line parameter is `--openPMD.dataPreparationStrategy`.
We could try unifying this, but there would be API breaking.